pkg/controller/plan/adapter/ovirt/builder.go

```diff
@@ -120,6 +120,37 @@ var osMap = map[string]string{
 	"windows_11":           "win11",
 }
 
+// Map of ovirt guest ids to instance type preference names.
+var osInstanceTypeMap = map[string]string{
```

Comment:
Can we try to make this dynamic?

Please note that instanceTypes and preferences carry a lot of labels which can be used for the mapping.

Reply:
> Can we try to make this dynamic?

+1
loading it from some config map would be nice

> but i do think we can query the list of preferences, iterate and check relevant name/annotation..

right, but with proper label(s) we can rather do it in a "k8s way" using a label selector
I also can't find a good label to use for selection though, for example for RHEL 7 we only have:
```yaml
  labels:
    app.kubernetes.io/component: templating
    app.kubernetes.io/managed-by: ssp-operator
    app.kubernetes.io/name: common-instancetypes
    app.kubernetes.io/part-of: hyperconverged-cluster
    app.kubernetes.io/version: 4.14.3
    instancetype.kubevirt.io/common-instancetypes-version: v0.3.3
    instancetype.kubevirt.io/os-type: linux
    instancetype.kubevirt.io/vendor: redhat.com
```

Reply:
added config maps

Reply:
Would adding an identifier similar to the ones used by osinfo-db help?

Reply:
yes, as you can already see in previous comment, we already using `os.template.kubevirt.io/<os>` when the `os` is mapped from vsphere/ovirt to the osinfo. that would've saved us from duplication (which we can remove later on and use the config map introduced here).